conversant/prompt_chatbot.py

```diff
@@ -128,11 +188,23 @@ def reply(self, query: str) -> Interaction:
         Returns:
             Interaction: Dictionary of query and generated LLM response
         """
+
         # The current prompt is assembled from the initial prompt,
         # from the chat history with a maximum of max_context_examples,
         # and from the current query
         current_prompt = self.get_current_prompt(query)
 
+        current_prompt_size = self.co.tokenize(current_prompt).length
+
+        if current_prompt_size > self.max_prompt_size:
+            self.chatbot_config[
```

Comment:
Should we avoid overriding the configured value here? The desired user-set value may not work at a current point in time, but might be okay later on.

Reply:
An alternative to this is re-computing a `allowed_max_context_examples` (didn't give much thought to the name yet) every time `current_prompt_size > max_prompt_size`, which shouldn't be too expensive since we're just doing arithmetic in `_update_max_context_examples`.

Reply:
About the number of calls I think we need 2 always, because we need one to see the amount of tokens before call .generate and we need the second one after so we can see the amount of tokens generated

Reply:
About always recomputing the max_context_examples, I was talking with Eddie, and I don’t know what is the best option in this case. Just updating the original value, since we are sending a warning, don’t look like a bad option. Thinking about  always recompute can be good option since we would return to the original value chosen, but  maybe it would be more confusing  constantly changes to this value, there would have to be warnings at each change. What do you think?

Reply:
In 
```python
self.prompt_size_history.append(
            self.co.tokenize(
                self.prompt.create_interaction_string(query, response)
            ).length
        )
```
we are tokenizing the interaction string:
"""
User: {query}
Bot: {response}
"""
If we set `return_likelihoods="GENERATION"` in the call to `co.generate`, we can retrieve the number of tokens for the response, but adding that to the rest of the interaction string may not be lead to the same token length as if we tokenize the entire interaction string.

So I agree with the second call of `co.tokenize()`!

Reply:
I think my preference is to recompute and warn at each time, for 3 reasons:
- We default back to the user desired config as soon as we are able to (instead of overriding the user's config)
- Related to above, we choose as large a context window as possible each time 
- I prefer to warn the user each time there's an instance of deviation from their desired config, so it's clear something is different each time.

Curious what do the others think!

Reply:
Thanks! I agree with you both, I think now all the suggestions are implemented :)